front/lib/api/assistant/legacy_agent.ts

```diff
@@ -548,6 +541,54 @@ async function* runAction(
           return;
       }
     }
+  } else if (isProcessConfiguration(action)) {
+    const eventStream = runProcess(auth, {
+      configuration,
+      actionConfiguration: action,
+      conversation,
+      userMessage,
+      agentMessage,
+      rawInputs,
+    });
+
+    for await (const event of eventStream) {
+      switch (event.type) {
+        case "process_params":
+          yield event;
+          break;
+        case "process_error":
+          yield {
+            type: "agent_error",
+            created: event.created,
+            configurationId: configuration.sId,
+            messageId: agentMessage.sId,
+            error: {
+              code: event.error.code,
+              message: event.error.message,
+            },
+          };
+          return;
+        case "process_success":
+          yield {
+            type: "agent_action_success",
+            created: event.created,
+            configurationId: configuration.sId,
+            messageId: agentMessage.sId,
+            action: event.action,
+          };
+
+          // We stitch the action into the agent message. The conversation is expected to include
+          // the agentMessage object, updating this object will update the conversation as well.
+          agentMessage.action = event.action;
+          break;
+
+        default:
```

Comment:
We can probably just assert, right? For sure, we don't need the [IIFE](https://developer.mozilla.org/en-US/docs/Glossary/IIFE).

Reply:
Yep legacy code, moved to assertNever